### DISCUSSION TITLE
CascadingBlazoredModal

### DISCUSSION THREAD

**Discussion by GiffenGood** [Blazored/Modal#322] [Q&A] [answered]
Is there any technical reason why CascadingBlazoredModal  must wrap the router?  Could I, for example, use CascadingBlazoredModal  to wrap the MainLayout?   

The reason I want to do this is that blazored modal does not work if I try to show a Telerik control in a dialog.  Telerik controls require that the MainLayout use their layout.  Since CascadingBlazoredModal  is defined outside of the Mainlayout as prescribed in the docs, the Telerik controls throw the following exception:

A Telerik component on the requested view requires a TelerikRootComponent to be added to the root of the MainLayout component of the app. Read more at: https://docs.telerik.com/blazor-ui/getting-started/what-you-need#project-configuration

As an experiment, I have wrapped my MainLayout with CascadingBlazoredModal  and it seems to work, but I want to be sure I am not backing myself into any corners.

Thanks,

MB

**Comment by chrissainty** [comment, marked as answer]
Hey @GiffenGood. As you've found, there is no issue with what you've done and the modal will work fine. The reason we state wrapping the router is that if an app is using different layout components for different areas, then the modal will continue to work.
